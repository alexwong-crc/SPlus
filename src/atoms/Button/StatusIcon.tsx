import React from "react";
import styled from "styled-components";
import { StatusProps } from "./Button";
import Icon from "../Icon";

interface IStatusIcon {
  status?: StatusProps;
}

type IconNames = "spinner" | "check" | "times";

export const StyledIcon = styled.div<IStatusIcon>`
  position: absolute;
  transform: translateY(-200%);
  transition: 0.3s;
  &.slideInOut {
    animation-name: slideInOut;
    animation-duration: 1s;
  }

  @keyframes slideInOut {
    0% {
      transform: translateY(-200%);
    }
    25% {
      transform: translateY(0);
    }
    75% {
      transform: translateY(0);
    }
    100% {
      transform: translateY(-200%);
    }
  }
`;

StyledIcon.displayName = "StyledIcon";

const StatusIcon: React.SFC<IStatusIcon> = ({ status }: IStatusIcon) => {
  const renderIcon = (): IconNames => {
    switch (status) {
      case "success":
        return "check";
      case "failure":
        return "times";
      default:
        return "times";
    }
  };

  return (
    <>
      <StyledIcon
        id="StyledIcon"
        className={
          status === "success" || status === "failure" ? "slideInOut" : ""
        }
      >
        <Icon icon={renderIcon()} transform="grow-5" />
      </StyledIcon>
      {status === "loading" ? (
        <Icon icon="spinner" spin transform="grow-5" />
      ) : null}
    </>
  );
};

export default StatusIcon;
