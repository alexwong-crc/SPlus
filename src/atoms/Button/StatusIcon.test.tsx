import React from "react";
import { shallow } from "enzyme";
import StatusIcon, { StyledIcon } from "./StatusIcon";

describe("StatusIcon", () => {
  it("should match last snapshot", () => {
    const component = shallow(<StatusIcon />);
    expect(component).toMatchSnapshot();
  });

  it("should have className slideInOut if status is success", () => {
    const component = shallow(<StatusIcon status="success" />);
    expect(component.find("StyledIcon").props().className).toEqual(
      "slideInOut"
    );
  });

  it("should have className slideInOut if status is failure", () => {
    const component = shallow(<StatusIcon status="failure" />);
    expect(component.find("StyledIcon").props().className).toEqual(
      "slideInOut"
    );
  });

  it("should have empty string for className if status is undefined", () => {
    const component = shallow(<StatusIcon />);
    expect(component.find("StyledIcon").props().className).toEqual("");
  });

  it("should have empty string for className if status is loading", () => {
    const component = shallow(<StatusIcon status="loading" />);
    expect(component.find("StyledIcon").props().className).toEqual("");
  });
});

describe("StyledIcon", () => {
  it("should match snapshot", () => {
    const component = shallow(<StyledIcon />);
    expect(component).toMatchSnapshot();
  });
});
