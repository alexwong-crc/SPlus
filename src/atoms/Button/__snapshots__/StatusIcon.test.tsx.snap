// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusIcon should match last snapshot 1`] = `
<Fragment>
  <StyledIcon
    className=""
    id="StyledIcon"
  >
    <Icon
      icon="times"
      transform="grow-5"
    />
  </StyledIcon>
</Fragment>
`;

exports[`StyledIcon should match snapshot 1`] = `
.c0 {
  position: absolute;
  -webkit-transform: translateY(-200%);
  -ms-transform: translateY(-200%);
  transform: translateY(-200%);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0.slideInOut {
  -webkit-animation-name: slideInOut;
  animation-name: slideInOut;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
}

<div
  className="c0"
/>
`;
